presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcTableHandle.java

```diff
@@ -39,10 +42,11 @@
     private final String tableName;
     private final TupleDomain<ColumnHandle> constraint;
     private final OptionalLong limit;
+    private final Optional<List<ColumnHandle>> projection;
```

Comment:
Instead of having it as `Optional<List<ColumnHandle>>` can we maintain it as `List<ColumnHandle>` so when we create a `JdbcTableHandle` for the fist time we would pass all the columns then as we applyProjection we might reduce it

Reply:
As I understand `TableHandle` is not only acquired to run a sql query. Getting list of columns always could be a bit expensive.

The more important fact is that, without `Optional` I won't be able to tell if `applyProjection` was actually called. Having obsolete list of columns in table handle (instead of nothing) could lead to query failure or incorrect results.

Reply:
Making it an eagerly populated list makes sense from SELECT perspective. But, currenlty TableHandle is used for all sorts of other operation (drop, alter, ...).
It;s both "table relation" (for SELECT) and "table identifier" (for everything else).
Until we separate those two concepts, we need to live with compromises... like having Optional column list, because really List<Columns> doesn't make sense for a drop table call.

I stumbled into this independently recently. I do think we need to split the use of TableHandle into two(+) concepts so that "table relation" and "table identifier" are represented separately.
cc @kokosing @electrum @martint